### PR TITLE
Jlc/phone validation backend

### PR DESCRIPTION
**Pull request by johanseto**
## Description

Add basic backend with generate and validate endpoint for OTP in phone validation.

## Testing instructions

Checkout this branch in eox-nelp.

Checkout the nelp-edx-theme in to this branch. 
https://github.com/nelc/nelp-edx-theme/pull/174
Add the following settings:
``` python
MFE_CONFIG["PHONE_VALIDATION_API_URL"] = "/eox-nelp/api/phone-validation/v1"
```
1. Login with an account without phone_number in profile in admin.
image
2. Go to `/dashboard` and check the modal of phone confirmation is working.
image
3. Enter a test phone -number and check there is a log related in lms related the creation of the otp.
Peek 2024-03-22 14-24

5. Then you need the validation code, to get it you can go to cache and find it using `{username}-{phone_number}`. 
image
6. Add the code in the modal of phone_validation to continue the flow, check there is a log of validating, and the modal return with success.
Peek 2024-03-22 14-28

7. Reload `/dashboard`, verify that  phone_validation modal doesn't appear. Also in the admin profile phone_number has been updated.

image

## Additional information

Jira story:
https://edunext.atlassian.net/browse/FUTUREX-692

## Checklist for Merge

- [ ] Tested in a remote environment
- [ ] Updated documentation
- [ ] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

**Review by andrey-canon — CHANGES_REQUESTED** [review]

File: eox_nelp/tests/test_utils.py [hunk 1]
```diff
@@ -222,3 +225,22 @@ def test_invalid_input(self, input_value):
             - TypeError is raised
         """
         self.assertRaises(TypeError, camel_to_snake, input_value)
+
+
+def get_key_expiration(key):
```
[the remark this answers is not in the document]
Comment (johanseto) [reply eox_nelp/tests/test_utils.py]: 9d6711b3b0f8dadabd9674d54775b3aeeb746003

File: eox_nelp/tests/test_utils.py [hunk 2]
```diff
@@ -222,3 +225,22 @@ def test_invalid_input(self, input_value):
...
+    """Util to get the expiration time of a key cache in seconds."""
```
[the remark this answers is not in the document]
Comment (johanseto) [reply eox_nelp/tests/test_utils.py]: https://github.com/eduNEXT/eox-nelp/pull/144/commits/9d6711b3b0f8dadabd9674d54775b3aeeb746003

File: eox_nelp/tests/test_utils.py
[the remark this answers is not in the document]
Comment (johanseto) [reply eox_nelp/tests/test_utils.py:1]: 647eb5ae0889f41b159def43753126493af3eb91

**Review by andrey-canon — CHANGES_REQUESTED** [review]

File: eox_nelp/tests/utils.py [hunk 3]
```diff
@@ -70,8 +70,11 @@ def set_key_values(element):
 def get_cache_expiration_time(key):
     """
     Util to get the expiration time of a key cache in seconds.
-    Expected behavior:
-        - Return expiration emaining time of the key in seconds.
+    Args:
+        key <str>: bla blabla
```
Comment (andrey-canon) [inline eox_nelp/tests/utils.py]: :)

**Review by andrey-canon — APPROVED** [review]

**Pull request merged**